target/linux/mvebu/files-6.6/arch/arm64/boot/dts/marvell/armada-7040-rb5009.dts

```diff
@@ -0,0 +1,361 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "armada-7040.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "MikroTik RB5009";
+	compatible = "mikrotik,rb5009", "marvell,armada7040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x40000000>;
+	};
+
+	aliases {
+		led-boot = &led_user;
+		led-failsafe = &led_user;
+		led-running = &led_user;
+		led-upgrade = &led_user;
+	};
+
+	sfp_i2c: sfp-i2c {
+		compatible = "i2c-gpio";
+		sda-gpios = <&cp0_gpio1 0 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&cp0_gpio1 1 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&cp0_gpio1 28 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: user {
+			label = "green:user";
+			gpios = <&cp0_gpio2 26 GPIO_ACTIVE_LOW>;
+		};
+
+		sfp {
+			label = "green:sfp";
+			gpios = <&cp0_gpio2 25 GPIO_ACTIVE_LOW>;
+		};
+
+		hdr1 {
+			label = "blue:hdr1";
+			gpios = <&cp0_gpio1 4 GPIO_ACTIVE_LOW>;
+		};
+
+		hdr2 {
+			label = "blue:hdr2";
+			gpios = <&cp0_gpio2 19 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	sfp: sfp {
+		compatible = "sff,sfp";
+		i2c-bus = <&sfp_i2c>;
+		mod-def0-gpios = <&cp0_gpio1 11 GPIO_ACTIVE_LOW>;
+		los-gpios = <&cp0_gpio1 2 GPIO_ACTIVE_HIGH>;
+		tx-fault-gpios = <&cp0_gpio1 6 GPIO_ACTIVE_HIGH>;
+		tx-disable-gpios = <&cp0_gpio1 5 GPIO_ACTIVE_HIGH>;
+		rate-select0-gpios = <&cp0_gpio1 3 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+
+	pinctrl-0 = <&uart0_pins>;
+	pinctrl-names = "default";
+};
+
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				compatible = "mikrotik,routerboot-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				label = "MikroTik";
+				reg = <0x0 0xfe0000>;
+
+				hard_config: hard_config {
+					read-only;
+
+					nvmem-layout {
+						compatible = "mikrotik,routerboot-nvmem";
+
+						macaddr_hard: base-mac-address {
+							#nvmem-cell-cells = <1>;
+						};
+					};
+				};
+
+				soft_config {
+				};
+
+				dtb_config {
+					read-only;
+				};
+			};
+
+			partition@b0000 {
+				label = "RouterBOOT";
```

Comment:
Could we please make it more clear that this partition is primary bootloader (only)? Perhaps: `bootloader2`?
There are many other `RouterBoot|routerboot` partitions in tree, and they are all both bootloaders+configs.

<details><summary><code>git grep -B2 -i '"routerboot";' master -- target/linux/{ath79,ipq40xx,ramips}</code></summary>

```
master:target/linux/ath79/dts/ar9342_mikrotik_routerboard-911g.dtsi-
master:target/linux/ath79/dts/ar9342_mikrotik_routerboard-911g.dtsi-			partition@0 {
master:target/linux/ath79/dts/ar9342_mikrotik_routerboard-911g.dtsi:				label = "routerboot";
--
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-16m-nor.dtsi-
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-16m-nor.dtsi-			partition@0 {
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-16m-nor.dtsi:				label = "RouterBoot";
--
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-951x-2hnd.dtsi-
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-951x-2hnd.dtsi-			partition@0 {
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-951x-2hnd.dtsi:				label = "routerboot";
--
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-sxt-5nd-r2.dts-
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-sxt-5nd-r2.dts-			partition@0 {
master:target/linux/ath79/dts/ar9344_mikrotik_routerboard-sxt-5nd-r2.dts:				label = "RouterBoot";
--
master:target/linux/ath79/dts/qca9533_mikrotik_routerboard-16m.dtsi-
master:target/linux/ath79/dts/qca9533_mikrotik_routerboard-16m.dtsi-			partition@0 {
master:target/linux/ath79/dts/qca9533_mikrotik_routerboard-16m.dtsi:				label = "RouterBoot";
--
master:target/linux/ath79/dts/qca9556_mikrotik_routerboard-wap-g-5hact2hnd.dts-
master:target/linux/ath79/dts/qca9556_mikrotik_routerboard-wap-g-5hact2hnd.dts-			partition@0 {
master:target/linux/ath79/dts/qca9556_mikrotik_routerboard-wap-g-5hact2hnd.dts:				label = "RouterBoot";
--
master:target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi-
master:target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi-			partition@0 {
master:target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi:				label = "routerboot";
--
master:target/linux/ath79/dts/qca9558_mikrotik_routerboard-96x.dtsi-
master:target/linux/ath79/dts/qca9558_mikrotik_routerboard-96x.dtsi-			partition@0 {
master:target/linux/ath79/dts/qca9558_mikrotik_routerboard-96x.dtsi:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-cap-ac.dts-				#address-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-cap-ac.dts-				#size-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-cap-ac.dts:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-hap-ac2.dts-				#address-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-hap-ac2.dts-				#size-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-hap-ac2.dts:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-sxtsq-5-ac.dts-				#address-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-sxtsq-5-ac.dts-				#size-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-sxtsq-5-ac.dts:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-wap-ac.dtsi-				#address-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-wap-ac.dtsi-				#size-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4018-wap-ac.dtsi:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-hap-ac3-lte6-kit.dts-			partition@80000 {
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-hap-ac3-lte6-kit.dts-				compatible = "mikrotik,routerboot-partitions";
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-hap-ac3-lte6-kit.dts:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-hap-ac3.dts-				#address-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-hap-ac3.dts-				#size-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-hap-ac3.dts:				label = "RouterBoot";
--
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-lhgg-60ad.dts-				#address-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-lhgg-60ad.dts-				#size-cells = <1>;
master:target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-lhgg-60ad.dts:				label = "RouterBoot";
--
master:target/linux/ramips/dts/mt7621_mikrotik.dtsi-
master:target/linux/ramips/dts/mt7621_mikrotik.dtsi-			partition@0 {
master:target/linux/ramips/dts/mt7621_mikrotik.dtsi:				label = "RouterBoot";
```

</details>

Reply:
All of those other occurrences are basically wrong and should be renamed as RouterBOOT is the bootloader.

Reply:
> RouterBOOT is the bootloader

Technicalities, but, I'm not so sure. primary bootloader is the fwf file from RouterOS squashfs, less header. Backup bootloader has extras.
There is a stub somewhere that checks if NOR soft config is set for backup bootloader, or if reset button is held while power applied. My guess is that is part of the extras in backup bootloader.
If primary bootloader was _the bootloader_, device would still work if you erased backup bootloader.

Reply:
Oh, I think we misunderstood each other.
This is just the primary instance of RouterBOOT, there is a backup one as well after ATF.
On this device I think they put FWF decompression and everything else directly into ATF.

> What would you suggest we rename our historic in-tree routerboot partitions to? Being all the space used by RouterBoot (bootloaders, plus the settings they need to function)?

Well, I just named the "container" partition "MikroTik" as I dont have a good name to use for it.

> I would be happy if you gave RouterBoot a suffix here?

`RouterBOOT-primary` and `RouterBOOT-backup`?